parcels/field.py

```diff
@@ -1545,18 +1553,76 @@ def __init__(self, filename, dimensions, indices, netcdf_engine, timestamp=None,
         self.interp_method = interp_method
         self.data_full_zdim = data_full_zdim
         self.field_chunksize = field_chunksize
+        self.chunk_mapping = None
+        self.rechunk_callback_fields = rechunk_callback_fields
 
     def __enter__(self):
         if self.netcdf_engine == 'xarray':
             self.dataset = self.filename
             return self
+
+        # ==== check-opening ==== #
+        try:
+            self.dataset = xr.open_dataset(str(self.filename), decode_cf=True, engine=self.netcdf_engine, chunks={})
+            self.dataset['decoded'] = True
+        except:
+            logger.warning_once("File %s could not be decoded properly by xarray (version %s).\n         It will be opened with no decoding. Filling values might be wrongly parsed."
+                                % (self.filename, xr.__version__))
+            self.dataset = xr.open_dataset(str(self.filename), decode_cf=False, engine=self.netcdf_engine, chunks={})
+            self.dataset['decoded'] = False
+
+        init_chunk_dict = {}
+        if isinstance(self.field_chunksize, dict):
+            init_chunk_dict = self.field_chunksize
+        elif isinstance(self.field_chunksize, tuple):
+            spatStart = 0
+            if 'time' in self.dimensions and self.dimensions['time'] in self.dataset.dims:
+                if(len(self.dimensions) > 2):
+                    init_chunk_dict[self.dimensions['time']] = self.field_chunksize[0]
+                    spatStart += 1
+                else:
+                    init_chunk_dict[self.dimensions['time']] = 1
+            else:
+                raise RuntimeWarning("Did not find {} in NetCDF dims - Please specify field_chunksize as dictionary for NetCDF dimension names.".format(self.dimensions['depth']))
```

Comment:
Do we need a bit more guidance _how_ to specify this `field_chuncksize`? E.g. link to a tutorial or FAQ (we can do that later, once we created the tutorial)?

Reply:
I can make a little example there for each case. And when we updated the documentation (https://github.com/OceanParcels/parcels/issues/720), then we should also update these messages.